src/tic_tac_toe/board.clj

```diff
@@ -1,11 +1,114 @@
-(ns tic-tac-toe.board)
+(ns tic-tac-toe.board
+  (:require [tic-tac-toe.marks :refer :all]))
 
-(def initial-board [1 2 3 4 5 6 7 8 9])
+(defn make-initial-board
+  [grid-size]
+  (into [] (map str (range 1 (+ (* grid-size grid-size) 1)))))
 
 (defn present-board
-  [grid-size board-values]
-  (print (clojure.string/join (apply concat (interpose ["\n"] (partition grid-size (map (fn [x] (str x " ")) (into [] board-values)))))) "\n"))
+  [board-values]
+  (print (clojure.string/join (apply concat (interpose ["\n"] (partition (int (Math/sqrt (count board-values))) (map (fn [x] (str x " ")) (into [] board-values)))))) "\n"))
 
 (defn place-mark
-  [position mark]
-  (assoc initial-board position mark))
+  [board position mark]
+  (assoc board position mark))
+
+(defn eq-mark-one-or-two
+  [mark]
+  (or (= mark player-one-mark) (= mark player-two-mark)))
+
+(defn is-full?
+  [board]
+  (every? (fn [mark] (eq-mark-one-or-two mark)) board))
+
+(defn top-row-winner?
+  [board]
+  (let [top-row [(get board 0) (get board 1) (get board 2)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
```

Comment:
You are repeating the above 3 lines quite often in the file, you could extract them into a reusable method, something like a more generic `winner?` or `winner-row?`, for example?

Reply:
Do you need the else-condition here? If the given condition is satisfied, the method returns true, otherwise false implicitly (normally)